src/rgw/rgw_iam_policy.h

```diff
@@ -446,6 +458,24 @@ struct Policy {
   Effect eval(const Environment& e,
 	      boost::optional<const rgw::auth::Identity&> ida,
 	      std::uint64_t action, const ARN& resource) const;
+
+  template <typename F>
+  bool has_conditional_p(const string& conditional, F p) const{
```

Comment:
This name is fine for me, but you could also just make it another overload of has_conditional(). 
If you find yourself back in this file, please add a space after "const".